R/config.R

```diff
@@ -5,6 +5,495 @@
+
+  api_response <- httr::GET(http_get_request)
+
+  if (iterating) {
+    result <- list()
+    result[["api_response"]] <- api_response
+    result[["type"]] <- type
+    result[["timestamp"]] <- timestamp
+
+    invisible(result)
+  } else {
+    openhub_download(api_response, save_folder_path, timestamp, type, unique_information=project_name, page=page)
+
+    invisible(api_response)
+  }
+}
+
+#' Downloads a single page of Analysis XML Response File.
+#'
+#' @description Downloads an XML response file from the "analysis" collection
+#' endpoint. The XML response file will return a single page containing the
+#' single analysis item. The XML response file is saved to disk using the
+#' function \code{\link{openhub_download}} with the file name
+#' "analysis_projectname_unixtimestamp.xml", where "projectname" is the
+#' sanitized version of the name of the project from `project_name`, and
+#' "unixtimestamp" is the integer unix timestamp derived from the most recent
+#' time this analysis collection was updated for the project ("updated_at"
+#' field in the analysis collection).
+#'
+#' @param token Your OpenHub API token.
+#' @param save_folder_path A folder path to save the downloaded XML page "as-is".
+#' @param project_id The unique ID of the project (e.g. 3562).
+#' @param project_name The unique name of the project (e.g. "Apache Tomcat").
+#' @return The single page XML response file that contains a single analysis item.
+#' @references For analysis collection details, see \url{https://github.com/blackducksoftware/ohloh_api/blob/main/reference/analysis.md}.
+#' @export
+openhub_api_analyses <- function(token, save_folder_path, project_id, project_name) {
+
+  analysis_collection_site_start <- "https://www.openhub.net/p/"
+
+  analysis_collection_site_end <- "/analyses/latest.xml"
+
+  type <- "analysis"
+
+  timestamp <- NULL
+
+  http_get_request <- paste0(analysis_collection_site_start, URLencode(project_id), analysis_collection_site_end, "?api_key=", token)
+
+  api_response <- httr::GET(http_get_request)
+
+  xmlDoc <- XML::xmlParse(api_response, validate=F)
+  root <- XML::xmlRoot(xmlDoc)
+  status <- XML::xmlValue(root[[1]]) # the value of <status>
+  returnItems <- root[[2]] # <result>
+  if (status == "success") {
+    updated_at <- XML::xmlValue(returnItems[[1]][[4]]) # value of <analysis><updated_at>
+    timestamp <- as.integer(as.POSIXct(updated_at, format="%Y-%m-%dT%H:%M:%SZ", tz="UTC"))
+  } else if (status == "Not Found") {
+    warning(paste0("openhub_api_analyses: No current analysis collection found for project name: ", project_name))
+  } else {
+    warning(paste0("openhub_api_analyses: ", status)) # prints the status warning message
+  }
+
+  if (!is.null(timestamp)) {
+    openhub_download(api_response, save_folder_path, timestamp, type, unique_information=project_name)
+  }
+
+  invisible(api_response)
```

Comment:
what is the purpose of this function? first time seeing it being used

Reply:
The `invisible()` function just like a `return()`, but it doesn't print the output of the function if a call to the function isn't assigned to a variable. This will be removed as there is no need for the API functions to return the API response files as they're downloaded.